org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend

```diff
@@ -3234,6 +3349,144 @@ class RenameParams extends TextDocumentPositionParams {
 	}
 }
 
+@Beta
+@JsonRpcData
+class SemanticTokensLegend {
+
+	/**
+	 * The token types that the client supports.
+	 */
+	@NonNull
+	List<String> tokenTypes;
+
+	/**
+	 * The token modifiers that the client supports.
+	 */
+	@NonNull
+	List<String> tokenModifiers;
+
+	new(@NonNull List<String> tokenTypes, @NonNull List<String> tokenModifiers) {
+		this.tokenTypes = Preconditions.checkNotNull(tokenTypes, 'tokenTypes')
+		this.tokenModifiers = Preconditions.checkNotNull(tokenModifiers, 'tokenModifiers')
+	}
+
+}
+
+@Beta
+@JsonRpcData
+class SemanticTokensServerFull {
+
+	/**
+	* The server supports deltas for full documents.
+	*/
+	Boolean delta
+
+	new() {
+	}
+
+	new(Boolean delta) {
+		this.delta = delta
+	}
+}
+
+@Beta
+@JsonRpcData
+class SemanticTokensOptions {
+
+	/**
+	 * The legend used by the server
+	 */
+	@NonNull
+	SemanticTokensLegend legend
+
+	/**
+	 * Server supports providing semantic tokens for a specific range
+	 * of a document.
+	 */
+	Either<Boolean, Object> range
+
+	/**
+	 * Server supports providing semantic tokens for a full document.
+	 */
+	Either<Boolean, SemanticTokensServerFull> full
+
+	new(@NonNull SemanticTokensLegend legend) {
+		this.legend = Preconditions.checkNotNull(legend, 'legend')
+	}
+
+	new(@NonNull SemanticTokensLegend legend, Boolean full) {
+		this(legend)
+		this.full = full
+	}
+
+	new(@NonNull SemanticTokensLegend legend, SemanticTokensServerFull full) {
+		this(legend)
+		this.full = full
+	}
+
+	new(@NonNull SemanticTokensLegend legend, Boolean full, Boolean range) {
+		this(legend)
+		this.full = full
+		this.range = range
+	}
+
+	new(@NonNull SemanticTokensLegend legend, SemanticTokensServerFull full, Boolean range) {
+		this(legend)
+		this.full = full
+		this.range = range
+	}
+
+}
+
+@Beta
+@JsonRpcData
+class SemanticTokensRegistrationOptions extends TextDocumentRegistrationOptions  {
```

Comment:
please add `WorkDoneProgressOptions`

Reply:
This class already extends TextDocumentRegistrationOptions, is there any other way?

Reply:
I see your PR merged on master now :sweat_smile: